mllib/src/main/scala/org/apache/spark/ml/Transformer.scala

```diff
@@ -90,7 +90,7 @@ abstract class UnaryTransformer[IN, OUT, T <: UnaryTransformer[IN, OUT, T]]
    * account of the embedded param map. So the param values should be determined solely by the input
    * param map.
    */
-  protected def createTransformFunc: IN => OUT
+  protected val createTransformFunc: (T, IN) => OUT
```

Comment:
This change is to make non-code-generated evaluation of `ScalaUDF` works with `RowEncoder` due to runtime mirror problem. I am thinking is it possible that we only support code-generated evaluation of `ScalaUDF`? Then we can avoid this change to `Transformer`. What you think? @rxin @davies Can you give some suggestions? Thanks!